### PR TITLE
Fixes #12859 - trigger autosuggesting even when the IP is set

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: app/assets/javascripts/host_edit.js [hunk 1]
```diff
@@ -547,12 +547,17 @@ $(document).on('change', '.interface_domain', function () {
...
-  clearError($(this).closest('fieldset').find('.interface_ip'));
+  var ip_field = $(this).closest('fieldset').find('.interface_ip');
+  clearError(ip_field);
+  ip_field.val('');
   interface_subnet_selected($(this).closest('fieldset').find('select.interface_subnet'));
   e.preventDefault();
 });
 
 $(document).on('change', '.interface_subnet', function () {
+  var ip_field = $(this).closest('fieldset').find('.interface_ip');
```
Comment (dLobatog) [inline app/assets/javascripts/host_edit.js]: Since this code is the same as the change above, how about extracting it to a function `clearIPField`?

**Comment by dLobatog** [comment]
@ares Works well, can you also add an integration test to clarify that the IP address must be empty in order to put a new one using 'suggest new ip'? Hopefully we can include this in 1.10

**Comment by ares** [comment]
@dLobatog how would the test work? If I click the suggest new link it clears the IP and fills it again, I'm not sure how I'd test the blank value which is there for very short time. I could check that IP changes (which would require real smart proxy with DHCP) or remains the same (internal DB IPAM) but that is effectively the same as when no new IP was suggested. I think this would be better to test on JS level but we don't have such tests.

**Comment by ares** [comment]
[test] because of ActionView::Template::Error: PG::Error: ERROR:  deadlock detected

**Comment by dLobatog** [comment]
Merged as d03db507341885416e99f518c66f8931e5d42dab, thanks @ares!

**Pull request closed without merging**